library/core/src/androidMain/kotlin/com/attafitamim/krop/core/crop/ImageCropper.android.kt

```diff
@@ -41,14 +47,35 @@ suspend fun ImageCropper.crop(
     crop(maxResultSize) { cached?.toImageSrc(context) }
 }
 
-const val CacheDir = "krop_cache"
+/**
+ * Initiates a new crop session, cancelling the current one, if any.
+ * Suspends until a result is available (cancellation, error, success) and returns it.
+ * The resulting image will be scaled down to fit [maxResultSize] if provided.
+ * [dataSource] will be used as a source.
+ * Set [cacheBeforeUse] to false if you're certain that reopening it multiple times won't be a problem,
+ * true otherwise.
+ */
+suspend fun ImageCropper.crop(
+    dataSource: ImageStream,
+    context: Context,
+    maxResultSize: IntSize? = DefaultMaxCropSize,
+    cacheBeforeUse: Boolean = true
```

Comment:
Even with the comment, I don't understand the meaning of this variable.

Reply:
On android, the file is cached if we pass true. This feature is from the old library, and probably solves the issue when the file is not always available to use from the original uri (deleted or permission changed).

I'm not sure if we should add the same feature for other platforms. In one of the issues, we discussed the case where we need to save the state and restore it later. Without caching the file, we might lose data or access by the time we restore it. Caching it in the memory is also not a good idea, since some files might be very large.

Reply:
Hmm that makes sense. Ideally, I personally, would always prefer to pass a file that is within my internal app directory and is always present to circumvent all of this, but yeah if you use an uri it might make sense to cache it.